diff --git a/node_modules/lighthouse/core/lib/page-functions.js b/node_modules/lighthouse/core/lib/page-functions.js
index de6a1d2..978f85f 100644
--- a/node_modules/lighthouse/core/lib/page-functions.js
+++ b/node_modules/lighthouse/core/lib/page-functions.js
@@ -576,8 +576,9 @@ function createEsbuildFunctionWrapper() {
  */
 function getRuntimeFunctionName(fn) {
   const match = fn.toString().match(/function ([\w$]+)/);
-  if (!match) throw new Error(`could not find function name for: ${fn}`);
-  return match[1];
+  const fnName = match ? match[1] : fn.name;
+  if (!fnName) throw new Error(`could not find function name for: ${fn}`);
+  return fnName;
 }
 
 // We setup a number of our page functions to automatically include their dependencies.
